rfc/text/advanced/ap_10_rpc_progressive_calls.md

```diff
@@ -1,5 +1,249 @@
+   |                   |                       |
+   |  RESULT or ERROR  |                       |
+   |<------------------|                       |
+,--+---.            ,--+---.                ,--+---.
+|Caller|            |Dealer|                |Callee|
+`------'            `------'                `------'
+
+A *Caller* indicates its willingness to issue a progressive call by setting
+
+{align="left"}
+CALL.Options.progress|bool := true
+
+*Example.* Caller-to-Dealer `CALL`
+
+{align="left"}
+```json
+    [
+        48,
+        77245,
+        {
+            "progress": true
+        },
+        "com.myapp.get_country_by_coords",
+        [50.450001, 30.523333]
+    ]
+```
+
+If the *Callee* supports progressive calls, the *Dealer* will forward the *Caller's* willingness to send progressive calls by setting
+
+{align="left"}
+INVOCATION.Details.progress|bool := true
+
+
+*Example.* Dealer-to-Callee `INVOCATION`
+
+{align="left"}
+```json
+    [
+        68,
+        35224,
+        379,
+        {
+            "progress": true
+        },
+        [50.450001, 30.523333]
+    ]
+```
+
+A call invocation MUST *always* end in a *normal* `CALL` without `"progress": true` option, or explicitly set `"progress": false` which is default.
+
+**Progressive calls and shared registration**
+
+RPCs can have a multiple registrations (see `Shared Registration` feature) with different `<invocation_policies>`.
+In this case `progressive` `CALLs` can be routed to different *Callees*, what can lead to unexpected results.
+To bind `INVOCATIONs` to the same *Callee* *Caller* can specify `sticky` option during `CALL`
+
+{align="left"}
+CALL.Options.sticky|bool := true
+
+
+In this case *Dealer* must make a first `INVOCATION` based on `<invocation_policy>` and then route all next `progressive` 
+calls to the same *Callee*.
+
+If binding all ongoing `progressive` calls to the same *Callee* is not required, *Caller* can set `sticky` option to `FALSE`. 
+
+If `CALL.Options.sticky` is not specified it is treated like `TRUE`, so all `progressive`
+calls go to the same *Callee*.
+
+**Progressive Call Cancellation**
+
+If the original *Caller* is no longer available for some reason (has left the realm), then the *Dealer* will send 
+an `INTERRUPT` to the *Callee*. The `INTERRUPT` will have `Options.mode` set to `"killnowait"` to indicate to the 
+client that no response should be sent to the `INTERRUPT`.
+
+```
+[INTERRUPT, INVOCATION.Request|id, Options|dict]
+```
+Options:
+```
+INTERRUPT.Options.mode|string == "killnowait"
+```
+
+Progressive call cancellation, like progressive call result cancellation closes an important safety gap: 
+In cases where progressive calls are used to stream data from *Callers*, and network connectivity is unreliable, 
+*Callers* my often get disconnected in the middle of sending progressive data. This can lead to unneeded memory
+consumption on *Dealer* and *Callee* sides, because of storing runtime data about ongoing calls.
```

Comment:
for the Dealer and Callee, due to the need to store temporary metadata about ongoing calls.